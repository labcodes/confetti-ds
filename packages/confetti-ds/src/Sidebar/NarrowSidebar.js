import React from "react";
import PropTypes from "prop-types";

export const NavbarWithDividersContext = React.createContext(false);
export default class NarrowSidebar extends React.Component {
  static propTypes = {
    /** Components that will be rendered in the NarrowSidebar (Header, Item, Logotype, etc). */
    children: PropTypes.node.isRequired,
    /** In mobile environment, defines if the Sidebar is open. */
    isOpenOnMobile: PropTypes.bool,
    /** Defines if the rendered Sidebar has a vivid style. */
    isVivid: PropTypes.bool,
    /** Defines if the Sidebar has dividers. */
    withDividers: PropTypes.bool,
  };

  static defaultProps = {
    isVivid: false,
    isOpenOnMobile: false,
    withDividers: false,
  };

  render() {
    const { children, isOpenOnMobile, isVivid, withDividers } = this.props;

    return (
      <NavbarWithDividersContext.Provider value={withDividers}>
        <div
          className={`lab-narrow-sidebar
            ${isVivid ? " lab-narrow-sidebar--vivid" : ""}
            ${withDividers ? " lab-narrow-sidebar--with-dividers" : ""}
            ${isOpenOnMobile ? " lab-narrow-sidebar--open-on-mobile" : ""}
            `}
        >
          {children}
        </div>
      </NavbarWithDividersContext.Provider>
    );
  }
}
