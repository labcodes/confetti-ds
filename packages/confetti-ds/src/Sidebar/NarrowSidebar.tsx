import React, { ReactNode } from "react";

export const NavbarWithDividersContext = React.createContext(false);

export interface NarrowSidebarProps {
  /** Components that will be rendered in the NarrowSidebar (Header, Item, Logotype, etc). */
  children: ReactNode;
  /** In mobile environment, defines if the Sidebar is open. */
  isOpenOnMobile?: boolean;
  /** Defines if the rendered Sidebar has a vivid style. */
  isVivid?: boolean;
  /** Defines if the Sidebar has dividers. */
  withDividers?: boolean;
}

export default function NarrowSidebar({
  children,
  isOpenOnMobile = false,
  isVivid = false,
  withDividers = false,
}: NarrowSidebarProps) {
  return (
    <NavbarWithDividersContext.Provider value={withDividers}>
      <div
        className={`lab-narrow-sidebar
            ${isVivid ? " lab-narrow-sidebar--vivid" : ""}
            ${withDividers ? " lab-narrow-sidebar--with-dividers" : ""}
            ${isOpenOnMobile ? " lab-narrow-sidebar--open-on-mobile" : ""}
            `}
      >
        {children}
      </div>
    </NavbarWithDividersContext.Provider>
  );
}
