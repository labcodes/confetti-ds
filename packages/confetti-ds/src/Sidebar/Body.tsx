import React, { ReactNode } from "react";

import { NavbarWithDividersContext } from "./NarrowSidebar";

export default function Body({ children }: {children?: ReactNode}) {
  return (
    <NavbarWithDividersContext.Consumer>
      {(withDividers) => (
        <nav
          className={
            `lab-narrow-sidebar__body` +
            `${withDividers ? ` lab-narrow-sidebar__body--with-scroll` : ` ""`}`
          }
        >
          {children}
        </nav>
      )}
    </NavbarWithDividersContext.Consumer>
  );
}
