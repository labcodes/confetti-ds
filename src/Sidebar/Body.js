import React from "react";
import PropTypes from "prop-types";

import { NavbarWithDividersContext } from "./NarrowSidebar";

/**
 *
 * @param children
 * @returns {JSX.Element} Components that will be rendered in the Body.
 * @constructor
 */
export default function Body({ children }) {
  return (
    <NavbarWithDividersContext.Consumer>
      {(withDividers) => (
        <nav
          className={
            `lab-narrow-sidebar__body` +
            `${withDividers ? ` lab-narrow-sidebar__body--with-scroll` : ` ""`}`
          }
        >
          {children}
        </nav>
      )}
    </NavbarWithDividersContext.Consumer>
  );
}

Body.propTypes = {
  /** Components that will be rendered in the Body. */
  children: PropTypes.node,
};

Body.defaultProps = {
  children: undefined,
};
