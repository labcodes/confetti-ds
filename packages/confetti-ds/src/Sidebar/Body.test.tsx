import React from "react";
import renderer from "react-test-renderer";

import Body from "./Body";
import { NavbarWithDividersContext } from "./NarrowSidebar";

describe("Body", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<Body>Test content</Body>)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with dividers", async () => {
    const renderedComponent = renderer
      .create(
        <NavbarWithDividersContext.Provider value>
          <Body>Test content</Body>
        </NavbarWithDividersContext.Provider>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
