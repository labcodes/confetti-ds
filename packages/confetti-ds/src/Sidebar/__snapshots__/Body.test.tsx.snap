// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders with base props 1`] = `
<nav
  className="lab-narrow-sidebar__body """
>
  Test content
</nav>
`;

exports[`Body renders with dividers 1`] = `
<nav
  className="lab-narrow-sidebar__body lab-narrow-sidebar__body--with-scroll"
>
  Test content
</nav>
`;
